components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2460,6 +2462,24 @@ public Group createGroup(Group group, Map<String, Boolean> requiredAttributes)
         return group;
     }
 
+    private void validateUserstoreDomain(String domainName) throws CharonException {
+
+        UserStoreManager userStoreManager = carbonUM.getSecondaryUserStoreManager(domainName);
+        if (userStoreManager != null) {
+            // There is a userstore with the given domain name.
+            return;
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Invalid userstore domain: " + domainName + " found in the request");
+        }
```

Comment:
can;t we simply return from here as we are anyway throwing a charonException at line 2374?